### PR TITLE
fix: Consolidate test functionality of client sessions tests.

### PR DESCRIPTION
**Pull request by paulhazen**

### CODE REVIEW

**Review by andrew-hirata-playeveryware — APPROVED** [review]

File: Assets/Tests/Common/EOSTestBase.cs
Comment (paulhazen) [inline Assets/Tests/Common/EOSTestBase.cs:1]: Please squash the two commits into one - for the sake of greater clarity.

Comment (paulhazen) [reply Assets/Tests/Common/EOSTestBase.cs:1]: Done! Nice catch @paulhazen

File: Assets/Tests/PlayMode/ClientSessionTests.cs [hunk 1]
```diff
@@ -117,19 +115,31 @@ public IEnumerator FindByBucketIdAndJoin()
             sessionSearchHandle.Find(ref findOptions, null, (ref SessionSearchFindCallbackInfo data) => { searchResult = data; });
 
             yield return new WaitUntil(() => searchResult != null);
```
Comment (WispyMouse) [inline Assets/Tests/PlayMode/ClientSessionTests.cs:117]: Not changed in scope; I wonder if WaitUntil should have a timeout here, so it doesn't roll infinitely.

**Review by WispyMouse — CHANGES_REQUESTED** [review]
Running the test twice currently always hangs Unity. I think the TearDown in EOSTestBase isn't implemented correctly. Either this needs to be determined as being out of the scope of this PR or needs to be fixed, since we can't really use the tests as is.

**Comment by paulhazen** [comment]
> Running the test twice currently always hangs Unity. I think the TearDown in EOSTestBase isn't implemented correctly. Either this needs to be determined as being out of the scope of this PR or needs to be fixed, since we can't really use the tests as is.

Yeah, I noticed this as well. Out of scope as far as I can tell, as I experienced similar behavior with the tests in the first place.

**Review by WispyMouse — APPROVED** [review]
Changing to approve. A fix for the bug I identified is in this PR: https://github.com/PlayEveryWare/eos_plugin_for_unity/pull/811

**Pull request merged**